cmd/clusterctl/client/cluster/proxy.go

```diff
@@ -359,11 +359,15 @@ func listObjByGVK(c client.Client, groupVersion, kind string, options []client.L
 	objList.SetAPIVersion(groupVersion)
 	objList.SetKind(kind)
 
-	if err := c.List(ctx, objList, options...); err != nil {
+	resourceListBackoff := newReadBackoff()
+	if err := retryWithExponentialBackoff(resourceListBackoff, func() error {
+		return c.List(ctx, objList, options...)
+	}); err != nil {
 		if !apierrors.IsNotFound(err) {
 			return nil, errors.Wrapf(err, "failed to list objects for the %q GroupVersionKind", objList.GroupVersionKind())
 		}
```

Comment:
Q: will this error still be detected when the client-go is not found error is (probably?) wrapped in some kind of retry error?


Should this if be rather inside the retry anyway? (i.e. do we want to retry if the list call works but we simply can't find a resource)

Reply:
Might be I'm wrong, but List should never return `apierrors.IsNotFound`, right?

Reply:
Would make sense, I also don't know.
We should find out :)

Reply:
I think you can entirely drop `if !apierrors.IsNotFound(err) { ` and then we can get this merged